OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -71,7 +71,12 @@ public override void OnBecomingIdle(Actor self)
 			if (withSpriteBody.DefaultAnimation.CurrentSequence.Name != Info.IdleSequence)
 				withSpriteBody.DefaultAnimation.PlayRepeating(Info.IdleSequence);
 
-			base.OnBecomingIdle(self);
+			var thumper = self.World.FindActorsInCircle(self.CenterPosition, Info.MaxSearchRadius)
```

Comment:
This is the hack that I was referring to in the other comment thread: We built the attraction code specifically to gracefuly handle situations like this.  The attraction does not need to be tied directly to targeting, and so fixing that would produce cleaner code and more consistent behaviour than this "pick a thumper and move".